custom_components/kumo/climate.py

```diff
@@ -106,6 +109,24 @@ async def async_setup_entry(
         raise ConfigEntryNotReady("Kumo integration found no indoor units")
     async_add_entities(entities, True)
 
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        name="get_hvac_schedule",
+        schema=None,
+        func="get_hvac_schedule",
+        supports_response=SupportsResponse.ONLY,
+    )
+    platform.async_register_entity_service(
+        name="set_hvac_schedule",
+        schema={
+            vol.Required("entity_id"): str,
+            # TODO: This could be a lot tighter...
+            vol.Required("schedule"): vol.Schema({}, extra=vol.ALLOW_EXTRA),
```

Comment:
There are online JSON schema generators that will write a JSON schema given an example JSON body. They're not perfect but can generate a good starting point. Or a tool like Swagger can be used to create a JSON schema from scratch.

Reply:
Providing an endpoint on pykumo to get the schedule schema (and using it here) might be a good way to hide this detail away from hass-kumo.

Reply:
(see main comment about when we want to start enforcing the schema) Yeah, I tried a bit to get the schema set (partially just to learn about how Home Assistant handles these things). I ran into confusing issues with `voluptuous`, but can take another look if you'd like to see this.

Reply:
Even if it's not a full schema validation but just code (inside pykumo) checking that required stuff is present and in the expected format, with useful error messages returned if not, that would be something. OTOH that's the sort of thing schema validation is good at. But I haven't played with whatever validator HA is using.

So I guess my preference would be for a somewhat loosely-defined schema that validates just the minimum. I'm thinking just something to protect the code from crashing due to unanticipated values/formats. Later, if you're able to make a higher-level interface in future, it would be different endpoints and so the schema would be different.